ciongke/cmd/test-pr/main.go

```diff
@@ -0,0 +1,392 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	gogithub "github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+	"gopkg.in/yaml.v2"
+
+	"github.com/kubernetes/test-infra/ciongke/cmdlog"
+	"github.com/kubernetes/test-infra/ciongke/gcs"
+	"github.com/kubernetes/test-infra/ciongke/github"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/batch"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+var (
+	repoURL       = flag.String("repo-url", "", "URL of the repo to test.")
+	repoOwner     = flag.String("repo-owner", "", "Owner of the repo to test.")
+	repoName      = flag.String("repo-name", "", "Name of the repo to test.")
+	pr            = flag.Int("pr", 0, "Pull request to test.")
+	branch        = flag.String("branch", "", "Target branch.")
+	head          = flag.String("head", "", "Head SHA.")
+	resultsBucket = flag.String("results-bucket", "", "Bucket to store job results.")
+	sourceBucket  = flag.String("source-bucket", "", "Bucket to store PR source.")
+	namespace     = flag.String("namespace", "", "Namespace for created pods.")
+	workspace     = flag.String("workspace", "/workspace", "Where to checkout the repo.")
+
+	oauthSecretFile = flag.String("oauth-secret-file", "/etc/oauth/oauth", "Path to the file containing the GitHub OAuth secret.")
```

Comment:
As above, calling this "github token" aligns better with Github's UI than "oauth secret".